### PR TITLE
Recipe card delete

### PR DESCRIPTION
**Pull request by michelletsun**
Recipe card delete now changes the card library state. Added confirmation window popup, too.

### CODE REVIEW

File: meal-planning/src/components/recipe-card/RecipeCard.js [hunk 1]
```diff
@@ -58,6 +72,31 @@ const RecipeCard = ({ recipe }) => {
...
+          <DialogActions>
+            <Button onClick={handleCloseNo} color="primary">
+              No, thanks!
+            </Button>
+            <Button onClick={handleCloseYes} color="primary" autoFocus>
+              Yes, please!
+            </Button>
+          </DialogActions>
+        </Dialog>
+      </div>
```
Comment (elainetsun) [inline meal-planning/src/components/recipe-card/RecipeCard.js:99]: This should be its own component. see AddRecipeDialog as an example. Move all the dialog stuff into a new file and control the state in this file.

File: meal-planning/src/components/recipe-card/RecipeCard.js [hunk 2]
```diff
@@ -58,6 +72,31 @@ const RecipeCard = ({ recipe }) => {
...
+        >
+          <DialogTitle id="alert-dialog-title">{"Delete the recipe for " + recipe.name +"?"}</DialogTitle>
+          <DialogContent>
+            <DialogContentText id="alert-dialog-description">
+              The recipe will be permanently deleted.
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={handleCloseNo} color="primary">
+              No, thanks!
```
Comment (elainetsun) [inline meal-planning/src/components/recipe-card/RecipeCard.js:92]: Just Yes and No on the buttons

File: meal-planning/src/components/recipe-card/RecipeCard.js [hunk 3]
```diff
@@ -58,6 +72,31 @@ const RecipeCard = ({ recipe }) => {
...
+            </Button>
+            <Button onClick={handleCloseYes} color="primary" autoFocus>
```
Comment (elainetsun) [inline meal-planning/src/components/recipe-card/RecipeCard.js:94]: we dont need the autoFocus

File: meal-planning/src/components/recipe-card/RecipeCard.js [hunk 4]
```diff
@@ -10,22 +10,36 @@ import DeleteIcon from '@material-ui/icons/Delete';
...
 
   const favoritedColor = favorite ? 'red' : '#0000008a';
 
   const handleFavoriteClick = () => {
     setFavorite(!favorite);
   };
 
-  const handleDeleteClick = () => {
-    const card = document.querySelector(`div[data-recipe-id="${recipe.id}"]`);
+  const handleClickOpen = () => {
```
Comment (elainetsun) [inline meal-planning/src/components/recipe-card/RecipeCard.js:32]:
```suggestion
  const handleDeleteOpen = () => {
```

File: meal-planning/src/components/recipe-card/RecipeCard.js [hunk 5]
```diff
@@ -10,22 +10,36 @@ import DeleteIcon from '@material-ui/icons/Delete';
...
+    setOpen(true);
+  };
 
-    if (window.confirm(`Are you sure you want to delete ${recipe.name}?`)) {
-      card.remove();
-    }
+  const handleCloseNo = () => {
```
Comment (elainetsun) [inline meal-planning/src/components/recipe-card/RecipeCard.js:36]:
```suggestion
  const handleDeleteClose= () => {
```

File: meal-planning/src/components/recipe-card/RecipeCard.js [hunk 6]
```diff
@@ -10,22 +10,36 @@ import DeleteIcon from '@material-ui/icons/Delete';
...
+    setOpen(false);
+  };
+
+  const handleCloseYes = () => {
```
Comment (elainetsun) [inline meal-planning/src/components/recipe-card/RecipeCard.js:40]:
```suggestion
  const handleDelete = () => {
```

File: meal-planning/src/components/recipe-library/RecipesLibrary.js [hunk 7]
```diff
@@ -54,6 +52,13 @@ const RecipesLibrary = () => {
     setRecipes(recipes.concat(recipe));
   };
 
+  const removeCard = recipe => {
+    const currentIndex = recipes.indexOf(recipe);
```
Comment (Jesrogers) [inline meal-planning/src/components/recipe-library/RecipesLibrary.js:56]:
```suggestion
    const currentIndex = recipes.findIndex(currentRecipe => recipe === currentRecipe);
```

**Pull request merged**